pkg/server/tls/dynamic_source.go

```diff
@@ -118,44 +118,37 @@ func (f *DynamicSource) Start(ctx context.Context) error {
 
 	// channel which will be notified when the leaf certificate reaches 2/3 of its lifetime
 	// and needs to be renewed
```

Comment:
I logged the timings of the renewals and the events that occured and found that there is logic was broken.
After the timer got triggered, the next iteration re-used the old timer value and triggered instantaneous.
The correct behavior is to wait for a new `renewMoment` to be received instead, that is what this change does.

Reply:
I have not figured out what caused this error, it might be one of the following:
- the change in https://github.com/cert-manager/cert-manager/pull/6914/files#r1571283869
- the refactor done in https://github.com/cert-manager/cert-manager/pull/6784
- previous refactor done in https://github.com/cert-manager/cert-manager/pull/4243

It is clear that the logic now is more sound and that the tests that were added in https://github.com/cert-manager/cert-manager/pull/6784 helped to find this issue.

Reply:
To backport this fix, I think the best approach is to backport the test/ debugging steps that I used in this PR and see if previous versions are affected also.